implementations/rust/ockam/ockam_app/src/cli/mod.rs

```diff
@@ -1,5 +1,40 @@
+use crate::error::Error::Generic;
+use crate::Result;
 use ockam_core::env::get_env_with_default;
+use tracing::{error, info};
 
-pub(crate) fn cli_bin() -> crate::Result<String> {
-    Ok(get_env_with_default("OCKAM", "ockam".to_string())?)
+/// Return the ockam executable path either from the OCKAM env. variable or as `ockam`
+pub(crate) fn cli_bin() -> Result<String> {
+    let ockam_path = get_env_with_default("OCKAM", "ockam".to_string())?;
+    Ok(ockam_path)
+}
+
+/// Check that the OCKAM environment variable defines an absolute path
+/// Otherwise we might fail to run the ockam command when starting the desktop application from an unexpected path
+/// Check that the ockam command can at least be called with the `--version` option and log
+/// its version
+pub(crate) fn check_ockam_executable() -> Result<()> {
+    let ockam_path = cli_bin()?;
+    if ockam_path != *"ockam" && !ockam_path.starts_with('/') {
+        let message = format!("The OCKAM environment variable must be defined with an absolute path. The current value is: {ockam_path}");
+        error!(message);
+        return Err(Generic(message));
+    };
+
+    match duct::cmd!(ockam_path, "--version").stdout_capture().run() {
```

Comment:
I was trying to hack this a little bit and I found two things:
- the tauri app binary accepts and ignores the `--version` argument. It didn't return an error and continued with its normal execution
- when running `cargo tauri dev` with the hacked version (conflicting binaries), the app hangs and the `check_ockam_executable` function never returns.

For the second point, perhaps we should be more conservative and exit early if the process doesn't return in a given time.

```rust
let handle = cmd.start()?;
for ... {
    let is_finished = handle.try_wait()?;
    ...
}
```

Reply:
If we're not married to `duct`, `tokio::process` has [Command::kill_on_drop](https://docs.rs/tokio/latest/tokio/process/index.html#droppingcancellation) which we could layer with `tokio::time::timeout`.

I'm almost certain this can be accomplished and successful with `duct` without much trouble, just to be clear, so I'm just offering the option.

Reply:
Something to keep in mind is that at that point we don't have the tauri async runtime ready, and I'm not sure we can create an ad-hoc tokio runtime for this (haven't tried it though, perhaps it's also a little overkill).

Reply:
Thanks for testing @adrianbenavides!

> For the second point, perhaps we should be more conservative and exit early if the process doesn't return in a given time

I think that we don't have to do this anymore since [there shouldn't be anymore conflicts with application names now](https://github.com/build-trust/ockam/pull/5723).

Reply:
Perhaps then this PR doesn't add the value it originally intended?

Reply:
Yes it does because during our testing I had some issue with the fact that the `OCKAM` env variable was not defined as an absolute path. Then:

- it brings an additional check it is pointing to at least an executable responding to `--version` (which will not be the desktop app because of the renaming)
- it displays the `ockam` version in the logs which can be useful to diagnose a mismatch of versions between the command line application and the desktop application

That being said we should quickly work towards _not_ having to use the `ockam` executable at all :-).